prow/plank/controller.go

```diff
@@ -537,7 +536,8 @@ func (c *Controller) syncTriggeredJob(pj prowapi.ProwJob, pm map[string]coreapi.
 			WithField("from", prevState).
 			WithField("to", pj.Status.State).Info("Transitioning states.")
 	}
-	return c.patchProwjob(prevPJ, pj)
+	_, err := pjutil.PatchProwjob(c.prowJobClient, c.log, prevPJ, pj)
```

Comment:
We should be doing `prevPJ := *pj.DeepCopy()`, right?

https://github.com/kubernetes/test-infra/blob/abf96837e87dd252a277632e47239cdcd37b9467/prow/apis/prowjobs/v1/zz_generated.deepcopy.go#L163

Reply:
why do we care about `prevPJ` beyong this point?

Reply:
When we create it

Reply:
I feel like we don't necessarily need deepcopy since we do not care about the unexported fields? But doesn't hurt to make the change... will update tomorrow

Reply:
done